src/Microsoft.DotNet.Build.Tasks/FilterRuntimesFromSupports.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using NuGet.Frameworks;
+using NuGet.RuntimeModel;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class FilterRuntimesFromSupports
+    {
+        public static List<CompatibilityProfile> GetAllTfmRidPairs(string supportsFile)
+        {
+            RuntimeGraph supportsGraph = JsonRuntimeFormat.ReadRuntimeGraph(supportsFile);
+            List<CompatibilityProfile> tfmRidPairs = new List<CompatibilityProfile>();
```

Comment:
I was actually imagining this returning FrameworkRuntimePairs rather than trying to preserve the actual names.